### ISSUE TITLE
[feature] OpenGraph for URLs in Posts (show preview of links)

### ISSUE THREAD

**Issue by dyedgreen** [superseriousbusiness/gotosocial#1345]
### Is your feature request related to a problem ?

There is a long and cryptic URL that’s unclear to parse (eg with a uuid as the path)

### Describe the solution you'd like.

Fetch the open graph/ twitter cards meta tags for the associated page and display a preview card.

### Describe alternatives you've considered.

NONE

**Comment by tsmethurst** [comment]
Right, this is something we sort of have in mind to do in the future, but it's not an immediate priority. We also gotta figure out a way of doing it without contributing to the hug of death effect (loads of fedi servers all pinging the same website at once when a link gets shared by a big account, bringing the website down).

**Comment by dyedgreen** [comment]
Yeah, I think you’d want to have some sort of throttling / randomness where it only fetches the link after a random number of minutes within some range?

(or even better the originating instance should fetch the metadata and federate it; but that would require cooperation from a bunch of other instances)

**Comment by dyedgreen** [comment]
Out of curiosity, which files / directories should I be looking at if I’m interested in contributing this functionality? 😅

**Comment by tsmethurst** [comment]
> the originating instance should fetch the metadata and federate it

this is the most interesting way of doing it imo

> which files / directories should I be looking at if I’m interested in contributing this functionality?

honestly, there are lots of places that would need to be changed / added to to make this work, but we're not really that interested in having it be in the code yet anyways, since it adds more maintenance cost + surface for bugs, and we have a loaaaaaad of other stuff to be working on right now ;)

**Comment by filipesmedeiros** [comment]
Just as a reference, Mastodon does this: https://docs.joinmastodon.org/entities/PreviewCard

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/1345.
